pkg/rtc/room.go

```diff
@@ -417,7 +417,7 @@ func (r *Room) Join(participant types.LocalParticipant, requestSource routing.Me
 		"numParticipants", len(r.participants),
 	)
 
-	if participant.IsRecorder() && !r.protoRoom.ActiveRecording {
+	if participant.Kind() == livekit.ParticipantInfo_EGRESS && !r.protoRoom.ActiveRecording {
```

Comment:
I think we'll want a backwards compatibility mode here. so that older Egress that are not setting kind would still work

Reply:
I'll check, egress 1.8.0+ requires livekit 1.5.1+ so it might be ok

Reply:
1.8.0 is setting Kind